### PR TITLE
Redux

### PR DESCRIPTION
**Pull request by MarinaStepanchuk**
Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module05
Screenshot:

Screenshot_33
Screenshot_34

Deploy: [movie-search-rs.netlify.app](https://movie-search-rs.netlify.app/)
Done 16.04.2023 / deadline 17.04.2023
Score: 15 / 15

### CODE REVIEW

File: src/components/Form/Form.test.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+import { vitest, Mock } from 'vitest';
 import { InputName } from '../../constants/common.constants';
+import { useAppDispatch } from '../../hooks/redux';
+
+vitest.mock('../../hooks/redux');
 
 describe('Form', () => {
-  const onSubmit = vitest.fn();
+  const mockDispatch = vitest.fn();
+  (useAppDispatch as Mock).mockReturnValue(mockDispatch);
```
Comment (Ksarelto) [inline src/components/Form/Form.test.tsx]: u can mock it earlier:
```
vitest.mock('../../hooks/redux', () => ({
  useAppDispatch: vitest.fn(),
}));
```

File: src/components/Form/Form.tsx [hunk 2]
```diff
@@ -47,6 +44,9 @@ const Form = (props: IFormProps): JSX.Element => {
     reValidateMode: 'onSubmit',
   });
 
+  const { showMessage, sendCard } = critiquesSlice.actions;
```
Comment (Ksarelto) [inline src/components/Form/Form.tsx]: u can add that export of actions to your slice and then import them here

File: src/components/ItemMovie/ItemMovie.test.tsx [hunk 3]
```diff
@@ -1,30 +1,17 @@
...
 
 describe('Movies', () => {
-  it('should send fetch and render modal card', async () => {
-    render(<ItemMovie key={movieMock.id} movie={movieMock} />);
-
-    const card = screen.getByText(movieDetailsMock.name);
-    fireEvent.click(card);
+  it('should render card', async () => {
+    render(
+      <Provider store={store}>
```
Comment (Ksarelto) [inline src/components/ItemMovie/ItemMovie.test.tsx:10]: if you have some time try to mock react-redux package

File: src/components/ItemMovie/ItemMovie.tsx [hunk 4]
```diff
@@ -19,61 +13,26 @@ interface IItemMoviesProps {
 }
 
 const ItemMovie = ({ movie }: IItemMoviesProps): JSX.Element | null => {
-  const { id, name, poster, year, countries = [], rating } = movie;
+  const { idCard } = useAppSelector((state) => state.modalReducer);
+  const { showModal } = modalSlice.actions;
```
Comment (Ksarelto) [inline src/components/ItemMovie/ItemMovie.tsx]: the same with actions

File: src/components/ItemMovie/ItemMovie.tsx [hunk 5]
```diff
@@ -19,61 +13,26 @@ interface IItemMoviesProps {
...
-  const showModalCard = (e: React.MouseEvent<HTMLDivElement>): void => {
+  const showModalWindow = (e: React.MouseEvent<HTMLDivElement>): void => {
     const targetElement = e.target as HTMLElement;
 
-    if (targetElement.closest(`.${classes.item}`)) {
-      getMovieDetails(e.currentTarget.id);
-      setShowModal(true);
-    }
-  };
+    if (!targetElement.closest(`.${classes.item}`)) return;
```
Comment (Ksarelto) [inline src/components/ItemMovie/ItemMovie.tsx]: try to not use inline if, it is not very readable, bettr use common if block

File: src/components/SearchBar/SearchBar.tsx [hunk 6]
```diff
@@ -1,37 +1,28 @@
...
-  };
-
-  useEffect(() => () => localStorage.setItem(LocalStorageKeys.search, `${inputText.current}`), []);
-
-  const onSubmitForm = (e: React.ChangeEvent<HTMLFormElement>): void => {
-    e.preventDefault();
-    setSearchText(inputText.current);
-    onTextChange(inputText.current);
+import { useAppDispatch, useAppSelector } from '../../hooks/redux';
+import { searchTextSlice } from '../../redux/store/reducers/SearchTextSlice/searchTextSlice';
```
Comment (Ksarelto) [inline src/components/SearchBar/SearchBar.tsx:3]: reducers are not components, so u can name them using camelCase)

File: src/components/Form/Form.test.tsx (hunk 1 above)
Comment (MarinaStepanchuk) [reply src/components/Form/Form.test.tsx]: Thanks, it looks neater.

File: src/components/SearchBar/SearchBar.tsx (hunk 6 above)
Comment (MarinaStepanchuk) [reply src/components/SearchBar/SearchBar.tsx:3]: Thanks, I'll keep that in mind for the future.)

File: src/components/ItemMovie/ItemMovie.test.tsx (hunk 3 above)
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.test.tsx:10]: I don't have time or desire to do it yet, maybe later).

File: src/components/ItemMovie/ItemMovie.tsx (hunk 5 above)
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.tsx]: ok, corrected.

File: src/components/Form/Form.tsx (hunk 2 above)
Comment (MarinaStepanchuk) [reply src/components/Form/Form.tsx]: replaced

File: src/components/ItemMovie/ItemMovie.tsx (hunk 4 above)
Comment (MarinaStepanchuk) [reply src/components/ItemMovie/ItemMovie.tsx]: replaced

**Pull request merged**